### PR TITLE
[VFS-778] Bug fixed: Returned channel will never be null

### PR DESCRIPTION
**Pull request by manishbansal8843**
https://issues.apache.org/jira/browse/VFS-778

### CODE REVIEW

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java
Comment (garydgregory) [inline commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java:158]: Javadoc it not aligned.

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java
Comment (garydgregory) [inline commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java:166]: Use `final` where possible.

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java
Comment (garydgregory) [inline commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java:171]: Fix formatting.

File: commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java
Comment (garydgregory) [inline commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/sftp/SftpFileSystem.java:135]: Fix formatting.

**Comment by manishbansal8843** [comment]
> This PR does not contain a unit test.

This PR only refactored some code. And i think, this complete file does not have any Unit tests. Please correct me if I am wrong.

**Comment by manishbansal8843** [comment]
@Gagravarr Can you please merge this fix if its ok? We are using the build from my patch branch to avoid this issue :)

**Comment by manishbansal8843** [comment]
@garydgregory i tried pulling some more dirt on this issue. This issue was fixed as part of #36 . However, code merging did not happen properly. Not sure how. The PR changes were [these](https://github.com/apache/commons-vfs/pull/36/files#diff-a670e11a4376db2a2bc2acc6c90a0e5a). However, the changes [merged ](https://github.com/apache/commons-vfs/commit/d9a5529b39ebffc3f59414e7af82178be58c019d#diff-a670e11a4376db2a2bc2acc6c90a0e5a)were different.

**Comment by garydgregory** [comment]
Hi @manishbansal8843 
This PR needs to be rebased on master and we still need a unit test. You can look at the package `org.apache.commons.vfs2.provider.sftp.test` for existing tests.

**Review by garydgregory — CHANGES_REQUESTED** [review]
Hi @manishbansal8843
This PR needs to be rebased on master and we still need a unit test. You can look at the package org.apache.commons.vfs2.provider.sftp.test for existing tests.

**Comment by manishbansal8843** [comment]
Hi @garydgregory Seems like you already merged the fix which was raised in PR #110 . I am not sure if this comes under any code of conduct, but i feel the author who raised an issue and PR first should get the credit.

**Comment by garydgregory** [comment]
Hello @manishbansal8843 
I did not consider this PR since it did not contain a test despite my two requests. On the other hand, the author of PR #110 added a test after my request. It's common for a test to take more time to create than it is to update the main code, but it is critical to do so, otherwise, as I am sure you know, a regression bug can be introduced by any future code change.

If you feel there is still a bug, or your changes to the main code are better because it fixes an issue on a related but different code path, then please rebase this PR on master, add a test, or create a new PR.
Thank you!

**Comment by manishbansal8843** [comment]
hi @garydgregory I understand that i did not add any test cases. hence, it was not merged. But that was not the only issue. This was a critical bug and as a maintainer, i was expecting some agility from your part to address this issue. I also pointed out that existing code did not have any test case (which is not an excuse to not add new tc). Had you pointed out it earlier that it was mandatory, i would have written it. But you mentioned the same after you already merged someone else PR addressing the same issue.

Had i spent effort without digging it out that existing PR has resolved this issue already, it would have wasted my time. As a maintainer, i expect you to be aware of all the PRs you are merging so that duplicate PRs can be avoided.

**Comment by garydgregory** [comment]
> hi @garydgregory I understand that i did not add any test cases. hence, it was not merged. But that was not the only issue. This was a critical bug and as a maintainer, i was expecting some agility from your part to address this issue. I also pointed out that existing code did not have any test case (which is not an excuse to not add new tc). Had you pointed out it earlier that it was mandatory, i would have written it. But you mentioned the same after you already merged someone else PR addressing the same issue.
> 
> Had i spent effort without digging it out that existing PR has resolved this issue already, it would have wasted my time. As a maintainer, i expect you to be aware of all the PRs you are merging so that duplicate PRs can be avoided.

Hello @manishbansal8843 

May I suggest that you adjust your expectations? I am a volunteer on this component as I am on all Apache projects I contribute to and as such provide my time for free and with my best intentions, always. I am not going to take the time to go and dissect what happened in what commit on what ticket and so on. Instead, I'll offer my support toward moving this issue forward. 

I would not say a test is 'mandatory' in each and every case, but I my approach is to prioritize my attention to PRs that do have tests while those that don't go to the back of the queue. If a PR does not have a test, I just ask for one and move on. 

Participating in the open source world for me, is not just creating PRs, throwing them over the wall, hoping that a project's maintainers match my expectations. For me, the process is a two way street, if I care enough about an issue, I create a PR, and manage my end of it, and try to participate in that project's process the best I can. Once you realize that there are folks on the other end that are just as busy as you are with day jobs, families, hobbies, and what nots, I think the process becomes less frustrating.

Keeping it simple from my end, re-reading https://github.com/apache/commons-vfs/pull/98#issuecomment-719949458 and summarizing: For me, no test means an issue goes to the back of the queue, regressions should be avoided. I don't micro manage or provide micro feedback on the process, this is not a corporate job for me, quite the opposite indeed ;-)

**Comment by manishbansal8843** [comment]
Hi @garydgregory ,

I understand your point of view. Hope you understood mine as well. Like you, i am also a volunteer. However, you having higher privilege as a member, does increase the expectations as well. But after reading your comment, i certainly feel my expectations are wrong. May be i need to get acquainted with the style of processes being followed here. Thanks.

**Pull request closed without merging**
